lib/SIL/SILVerifier.cpp

```diff
@@ -4345,6 +4345,40 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     }
     
     SILInstructionVisitor::visitSILBasicBlock(BB);
+
+    // Visit the instructions in the BB, to find holes in the lexical scopes.
+    // We use a fairly simple heuristic here (Explain heuristic).
+    llvm::DenseSet<const SILDebugScope *> AlreadySeenScopes;
+    if (BB->getParent()->getEffectiveOptimizationMode() !=
+        OptimizationMode::NoOptimization)
+      return;
+    if (BB->empty())
+      return;
+    SILInstruction &First = *BB->begin();
+    const SILDebugScope *LastSeenScope = First.getDebugScope();
+    assert(LastSeenScope && "Invalid top-level debug scope");
+    AlreadySeenScopes.insert(LastSeenScope);
+    for (SILInstruction &SI : *BB) {
+      auto *DS = SI.getDebugScope();
+      assert(DS && "Each instruction should have a debug scope");
+      if (!AlreadySeenScopes.count(DS)) {
+        AlreadySeenScopes.insert(DS);
+        LastSeenScope = DS;
+        continue;
```

Comment:
If you always continue in the `if` branch you might as well omit the `else`.

Reply:
Done